assignment/core/src/main/java/com/assignment/core/cipherservices/impl/AESAsymmetricCipherProvider.java

```diff
@@ -0,0 +1,52 @@
+package com.assignment.core.cipherservices.impl;
+
+import com.assignment.core.cipherservices.AsymmetricCipherProvider;
+import org.osgi.service.component.annotations.Component;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.util.Base64;
+
+@Component(service = AsymmetricCipherProvider.class, immediate = true)
+public class AESAsymmetricCipherProvider implements AsymmetricCipherProvider {
+    private static final String ALGORITHM = "AES";
+    private SecretKey secretKey;
+
+    public AESAsymmetricCipherProvider() throws Exception {
+        KeyGenerator keyGen = KeyGenerator.getInstance(ALGORITHM);
+        keyGen.init(256);
+        secretKey = keyGen.generateKey();
+    }
+
+    @Override
+    public String getAlgorithm() {
+        return ALGORITHM;
+    }
+
+    @Override
+    public String encrypt(String text) {
+        try {
+            Cipher cipher = Cipher.getInstance(ALGORITHM);
+            cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+            byte[] encryptedBytes = cipher.doFinal(text.getBytes());
+            return Base64.getEncoder().encodeToString(encryptedBytes);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Please check exception handling here.
`catch (Exception e) is too generalised` and I am not sure how `throw new RuntimeException(e)` serves any purpose.